src/Sentry/SentryOptions.cs

```diff
@@ -574,6 +574,16 @@ public double TracesSampleRate
         /// </remarks>
         public Func<TransactionSamplingContext, double?>? TracesSampler { get; set; }
 
+        /// <summary>
+        /// A customizable list of <see cref="TracePropagationTarget"/> objects, each containing either a
+        /// substring or regular expression pattern that can be used to control which outgoing HTTP requests
+        /// will have the <c>sentry-trace</c> and <c>baggage</c> headers propagated, for purposes of distributed tracing.
+        /// The default value is <c>null</c>, which indicates that the headers will be propagated to ALL targets.
+        /// To disable propagation completely, set this value to an empty collection.
+        /// </summary>
+        /// <seealso href="https://develop.sentry.dev/sdk/performance/#tracepropagationtargets"/>
+        public ICollection<TracePropagationTarget>? TracePropagationTargets { get; set; } = null;
```

Comment:
Other SDKs just do a plain list of Strings that can be either a String or a Regex.
This solution is nice because forces the user to tell, whether this is a String or a Regex, so the SDK does not need to handle it blindly.
Downside is more types, because of bundle size, always, trade-offs =D

Reply:
Bigger downside is that we couldn't take advantage of [`RegexOptions.Compiled`](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options#compiled-regular-expressions).  We'd be essentially re-parsing the regex pattern for every request.

I have adjusted the implementation to more closely resemble the others, but I'll still keep the custom type so I can build the compiled regex once on app startup.

Reply:
TIL compiled regexes, Dart is compiled in the ctor.